go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

```diff
@@ -538,15 +538,17 @@ func (zic *ZoneInterconnectHandler) addRemoteNodeStaticRoutes(node *corev1.Node,
 	addRoute := func(prefix, nexthop string) error {
 		logicalRouterStaticRoute := nbdb.LogicalRouterStaticRoute{
 			ExternalIDs: map[string]string{
-				"ic-node": node.Name,
+				"ic-node":               node.Name,
+				types.NetworkExternalID: zic.GetNetworkName(),
```

Comment:
So if this external ID wasn't already being set, what happens on upgrade to this change?  Wouldn't this create a new LRSR and the old one be left over completely unmanaged?

Reply:
fixed, please take a look.

Reply:
@jcaamano the implementation of CreateOrReplaceLogicalRouterStaticRouteWithPredicate() is to find the logical static route **from the specified logical router only** with the predicate, so your concern won't be a problem.

Reply:
@jcaamano fixed based on our discussion. please take a look